### DISCUSSION TITLE
Auto generate username from email

### DISCUSSION THREAD

**Discussion by sburk** [goauthentik/authentik#11887] [Q&A]
Hi,
I'm new to Authentik. I want to realize an enrollment that can perform the following:
user enters email, e.g. tom.master@company.com
Authentik username ist automatically generated as tom.master
In python it would be something like user.username = email.split("@")[0]
I found [this discussion](https://github.com/goauthentik/authentik/discussions/2107) but maybe there was some progress?
Can i realize that so far?

Thanks in advance!

**Comment by hobgoblina** [comment]
have you found a solution for this? I'm trying to do it the other way around - use username to populate the email attribute by appending `@example.com` to the end of it

**Comment by hobgoblina** [comment]
figured it out, created the following expression policy and attached it to the prompt stage of my enrollment flow. I'd expect the same would work for generating usernames via `split('@')[0]` like you described .

```
request.context['prompt_data']['email'] = request.context['prompt_data']['username'] + '@example.com'
return True
```
